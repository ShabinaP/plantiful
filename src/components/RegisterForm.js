import React, { useState } from "react";
import axios from "axios";
import Alert from "./Alert";
import { Form, FloatingLabel, Button } from "react-bootstrap";
import { ArrowRight } from "phosphor-react";
import "../styles/RegisterForm.css";

const initialState = {
	fields: {
		username: "",
		email: "",
		password: "",
		confirmPassword: "",
	},
	alert: {
		message: "",
		isSuccess: false,
	},
};

const RegisterForm = () => {
	const [fields, setFields] = useState(initialState.fields);
	const [alert, setAlert] = useState(initialState.alert);

	const handleInputChange = (e) => {
		setFields({ ...fields, [e.target.name]: [e.target.value] });
	};
	const handleSubmit = (e) => {
		e.preventDefault();
		setAlert({ message: "", isSuccess: false });
		if (fields.password !== fields.confirmPassword) {
			console.log(fields.password);
			console.log(fields.confirmPassword);
			setAlert({
				message: "Passwords do not match",
				isSuccess: false,
			});
		} else {
			axios
				.post("http://localhost:4000/auth/register", {
					username: fields.username,
					email: fields.email,
					password: fields.password,
				})
				.then((response) => {
					console.log(response);
					setAlert({
						message: "Registration Successful",
						isSuccess: true,
					});
				})
				.catch((error) => {
					console.log(error);
					setAlert({
						message: "Server error. Please try again later",
						isSuccess: false,
					});
				});
		}
	};
	return (
		<div className="form_wrapper">
			<div>
				<h2 className='signup_header'>Sign up to Plantiful</h2>
			</div>
			<div className="form_container">
				<Form onSubmit={handleSubmit}>
					<Form.Group className='mb-3' controlId='formBasicUsername'>
						<FloatingLabel
							controlId='floatingInput'
							label='Create Username'
							className='mb-3'
						>
							<Form.Control
							    className="w-50"
								type='textbox'
								name='username'
								placeholder="Create Username"
								value={fields.username}
								onChange={handleInputChange}
								required
							/>
						</FloatingLabel>
					</Form.Group>

					<Form.Group className='mb-3' controlId='formBasicEmail'>
						<FloatingLabel
							controlId='floatingInput'
							label='Enter your email address'
							className='mb-3'
						>
							<Form.Control
								type='email'
								name='email'
								placeholder='Enter your email address'
								value={fields.email}
								onChange={handleInputChange}
								className="w-50"
								required
							/>
						</FloatingLabel>
						<Form.Text className='text-muted'>
							We'll never share your email with anyone else.
						</Form.Text>
					</Form.Group>
					<Form.Group className='mb-3' controlId='formBasicPassword'>
						<FloatingLabel
							controlId='floatingInput'
							label='Create Password'
							className='mb-3'
						>
							<Form.Control
								type='password'
								className="w-50"
								placeholder='Create Password'
								data-testid='passwordinput'
								name='password'
								value={fields.password}
								onChange={handleInputChange}
								required
							/>
						</FloatingLabel>
					</Form.Group>
					<Form.Group className='mb-3' controlId='formBasicConfirmPassword'>
						<FloatingLabel
							controlId='floatingInput'
							label='Confirm Password'
							className='mb-3'
						>
							<Form.Control
								type='password'
								className="w-50"
								placeholder="Confirm Password"
								data-testid='confirmPassword'
								name='confirmPassword'
								value={fields.confirmPassword}
								onChange={handleInputChange}
								required
							/>
						</FloatingLabel>
					</Form.Group>
					<Button
						type='submit'
						id="submit"
						className='button-primary'
						data-testid='submitbutton'
						style={{ backgroundColor: "#55A356", border: "#013606"}}
					>
						Create account <ArrowRight size={20} />
					</Button>
					<p className="member_text">Already a member?<a href="/loginform" className="signin_link"> Sign in </a></p>
					<Alert message={alert.message} success={alert.isSuccess} />
				</Form>
				</div>
		</div>
	);
};

export default RegisterForm;
