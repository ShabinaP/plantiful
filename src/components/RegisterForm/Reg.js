import "./register.css";
import React, { useContext } from "react";
import { DataContext } from "../../dataContext/DataContext";
function Reg() {
  const { message, handleFieldChange, fields, handleSubmit } =
    useContext(DataContext);
  console.log(fields);
  return (
    <>
      <div id="login-box">
        <div className="left">
          <h1>Sign up</h1>
          <form onSubmit={handleSubmit}>
            <label htmlFor="username">Name</label>
            <input
              type="text"
              className="input-title"
              id="name"
              name="name"
              value={fields.name}
              onChange={handleFieldChange}
              placeholder="Enter your name"
              required
            />

            <label htmlFor="username">username</label>
            <input
              type="text"
              className="input-title"
              id="username"
              name="username"
              value={fields.username}
              onChange={handleFieldChange}
              placeholder="Enter username"
              required
            />
            <input
              type="email"
              className="input-title"
              id="email"
              name="email"
              value={fields.email}
              onChange={handleFieldChange}
              placeholder="Enter email address"
              required
            />

            <label htmlFor="password">Password</label>
            <input
              type="password"
              className="input-title"
              id="password"
              name="password"
              value={fields.password}
              onChange={handleFieldChange}
              placeholder="Create Password"
              required
            ></input>

            <label htmlFor="confirmpassword">Confirm Password</label>
            <input
              type="password"
              className="form_control input-title"
              id="confirmPassword"
              data-testid="confirmPassword"
              name="confirmPassword"
              value={fields.confirmPassword}
              onChange={handleFieldChange}
              placeholder="Confirm Password"
              required
            ></input>

            <button
              type="submit"
              className="button-primary btn btn-primary"
              data-testid="submitbutton"
            >
              Create account
            </button>
            {/* <Alert message={alert.message} success={alert.isSuccess} /> */}
          </form>
        </div>

        <div className="right">
          <span className="loginwith">
            Sign in with
            <br />
            social network
          </span>

          <button className="social-signin facebook">
            Log in with facebook
          </button>
          <button className="social-signin twitter">Log in with Twitter</button>
          <button className="social-signin google">Log in with Google+</button>
        </div>
        <div className="or">OR</div>
      </div>
    </>
  );
}

export default Reg;
