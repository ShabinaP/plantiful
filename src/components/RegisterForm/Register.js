import React,{useContext} from 'react'
import {DataContext} from '../../dataContext/DataContext'
export default function Register() {

    const {message,handleFieldChange, fields, handleSubmit} = useContext(DataContext)
    console.log(fields)
    return (
        <>
        <div className="mainclass flex"> 
        <div className="register user">
         
          <form className="form">
            <fieldset className="input-text-fields">
              <label htmlFor="name">
                Name:
                <input
                  type="text"
                  value={fields.name}
                  className="input-title"
                  name="name"
                  id="name"
                  onChange={handleFieldChange}
                />
              </label>
              <label htmlFor="name">
                username:
                <input
                  type="text"
                  value={fields.username}
                  className="input-title"
                  name="username"
                  id="username"
                  onChange={handleFieldChange}
                />
              </label>

              <label htmlFor="email">
                Email:
                <input
                  type="email"
                  htmlFor="email"
                  name="email"
                  value={fields.email}
                  onChange={handleFieldChange}
                />
              </label>
            
              <label htmlFor="password">
                Password:
                <input
                  type="password"
                  value={fields.password}
                  className="input-title"
                  name="password"
                  id="password"
                  onChange={handleFieldChange}
                />
              </label>            
              <label htmlFor="name">
                Confirm password:
                <input
                  type="password"
                  value={fields.confirmPassword}
                  className="input-title"
                  name="confirmPassword"
                  id="confirmPassword"
                  onChange={handleFieldChange}
                />
              </label>
            
              <button className="button" type="submit" onClick={handleSubmit}  >
                Submit
              </button>
              <p> {message && message} </p>
            </fieldset>
          </form>
        </div>
       </div>
         </>
            
    )
}
