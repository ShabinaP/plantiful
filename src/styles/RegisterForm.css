.signup_header {
    display: flex;
    justify-content: center;
    align-items: center;
    color: #013606;
    padding: 15px 10px;
    font-weight: 700;
}

.form_wrapper {
    height: 85vh;
    padding: 20px;
    background-color: #F9F5F1;
    width: 1470px;
}

.form_container {
    padding: 20px;
}

.member_text {
    padding: 12px 7px;
}
.signin_link {
    color: #013606;
}