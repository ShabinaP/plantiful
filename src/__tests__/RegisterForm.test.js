import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import RegisterForm from "../components/RegisterForm";

describe("Register Form component", () => {
	it("renders the basic fields", () => {
		const { asFragment } = render(<RegisterForm />);
        expect(asFragment()).toMatchSnapshot();
	});
	it("should validate email field", () => {
		render(<RegisterForm />);
		const email = screen.getByRole("textbox", { name: /email address/i });
		userEvent.type(email, "hello@plantiful.com");
		expect(email).toHaveValue("hello@plantiful.com");
	});
    it("should validate username field", () => {
        render(<RegisterForm />)
        const username = screen.getByLabelText(/username/i);
        userEvent.type(username, "PlantUser");
        expect(username).toHaveValue("PlantUser");
    })
    it("should validate password field", () => {
        render(<RegisterForm />);
        const password = screen.getByTestId(/passwordinput/i);
        userEvent.type(password, "Il0veP!ant$!");
        expect(password).toHaveValue("Il0veP!ant$!");
    });
    it("should validate confirm password field", () => {
        render(<RegisterForm/>);
        const confirmPassword = screen.getByTestId(/confirmpassword/i);
        userEvent.type(confirmPassword, "Il0veP!ant$!");
        expect(confirmPassword).toHaveValue("Il0veP!ant$!")
    })
    it("the form submits", () => {
        const onSubmit = jest.fn();
        const { getByTestId } = render(<RegisterForm onSubmit={onSubmit()}/>);
        userEvent.click(getByTestId("submitbutton"));
        expect(onSubmit).toHaveBeenCalled();
    })
});
