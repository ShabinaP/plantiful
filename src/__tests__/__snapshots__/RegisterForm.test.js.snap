// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Form component renders the basic fields 1`] = `
<DocumentFragment>
  <div
    class="form_wrapper"
  >
    <div>
      <h2
        class="signup_header"
      >
        Sign up to Plantiful
      </h2>
    </div>
    <div
      class="form_container"
    >
      <form
        class=""
      >
        <div
          class="mb-3"
        >
          <div
            class="mb-3 form-floating"
          >
            <input
              class="w-50 form-control"
              id="floatingInput"
              name="username"
              placeholder="Create Username"
              required=""
              type="textbox"
              value=""
            />
            <label
              for="floatingInput"
            >
              Create Username
            </label>
          </div>
        </div>
        <div
          class="mb-3"
        >
          <div
            class="mb-3 form-floating"
          >
            <input
              class="w-50 form-control"
              id="floatingInput"
              name="email"
              placeholder="Enter email"
              required=""
              type="email"
              value=""
            />
            <label
              for="floatingInput"
            >
              Enter your email address
            </label>
          </div>
          <small
            class="text-muted form-text"
          >
            We'll never share your email with anyone else.
          </small>
        </div>
        <div
          class="mb-3"
        >
          <div
            class="mb-3 form-floating"
          >
            <input
              class="w-50 form-control"
              data-testid="passwordinput"
              id="floatingInput"
              name="password"
              placeholder="Password"
              required=""
              type="password"
              value=""
            />
            <label
              for="floatingInput"
            >
              Create password
            </label>
          </div>
        </div>
        <div
          class="mb-3"
        >
          <div
            class="mb-3 form-floating"
          >
            <input
              class="w-50 form-control"
              data-testid="confirmPassword"
              id="floatingInput"
              name="confirmPassword"
              placeholder="Confirm Password"
              required=""
              type="password"
              value=""
            />
            <label
              for="floatingInput"
            >
              Confirm password
            </label>
          </div>
        </div>
        <button
          class="button-primary btn btn-primary"
          data-testid="submitbutton"
          id="submit"
          style="background-color: rgb(85, 163, 86); border: #013606;"
          type="submit"
        >
          Create account 
          <svg
            fill="currentColor"
            height="20"
            viewBox="0 0 256 256"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <line
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
              x1="40"
              x2="216"
              y1="128"
              y2="128"
            />
            <polyline
              fill="none"
              points="144 56 216 128 144 200"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="16"
            />
          </svg>
        </button>
        <p
          class="member_text"
        >
          Already a member?
          <a
            class="signin_link"
            href="/loginform"
          >
             Sign in 
          </a>
        </p>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
